### PR TITLE
[Docs] Added MongoDB persistence in how-tos

### PR DESCRIPTION
**Pull request by theofficialvedantjoshi**

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -135,6 +135,7 @@ nav:
           - Add persistence ("memory"): how-tos/persistence.ipynb
           - Manage conversation history: how-tos/managing-conversation-history.ipynb
           - Create custom checkpointer using Postgres: how-tos/persistence_postgres.ipynb
+          - How to create a custom checkpointer using MongoDB: how-tos/persistence_mongodb.ipynb
```
Comment (vbarda) [inline docs/mkdocs.yml:138]:
```suggestion
          - Create a custom checkpointer using MongoDB: how-tos/persistence_mongodb.ipynb
```

File: examples/persistence_mongodb.ipynb [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture --no-stderr\n",
+    "%pip install -U psycopg psycopg-pool langgraph pymongo"
```
Comment (vbarda) [inline examples/persistence_mongodb.ipynb:30]: please delete these `psycopg psycopg-pool` as they're not used in the notebook

File: examples/persistence_mongodb.ipynb [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "res"
+   ]
+  },
+  {
+   "attachments": {
```
Comment (vbarda) [inline examples/persistence_mongodb.ipynb:455]: is there a way to show this without a screenshot? just by showing these outputs programmatically?

File: examples/persistence_mongodb.ipynb [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+    "\n",
+    "\n",
+    "_set_env(\"OPENAI_API_KEY\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup model and tools for the graph"
```
Comment (vbarda) [inline examples/persistence_mongodb.ipynb:375]: are you using this at all in the notebook? it seems disconnected from the basic usage example

File: examples/persistence_mongodb.ipynb [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+    "checkpointer = MongoDBSaver(MongoClient(MONGO_URI), \"checkpoints_db\", \"checkpoints_collection\")\n",
+    "builder = StateGraph(int)\n",
+    "builder.add_node(\"add_one\", lambda x: x + 1)\n",
+    "builder.set_entry_point(\"add_one\")\n",
+    "builder.set_finish_point(\"add_one\")\n",
+    "graph = builder.compile(checkpointer=checkpointer)\n",
+    "config = {\"configurable\": {\"thread_id\": \"123\"}}\n",
+    "graph.get_state(config)\n",
+    "result = graph.invoke(3,config)\n",
+    "graph.get_state(config)"
```
Comment (vbarda) [inline examples/persistence_mongodb.ipynb:323]: could you also add a couple more example outputs for `list`, `put` etc? similar to postgres checkpointer docs

**Review by vbarda — COMMENTED** [review]
thanks for adding this! left a few small comments but overall looks good to me

File: examples/persistence_mongodb.ipynb (hunk 3 above)
Comment (theofficialvedantjoshi) [reply examples/persistence_mongodb.ipynb]: done

File: examples/persistence_mongodb.ipynb (hunk 5 above)
Comment (theofficialvedantjoshi) [reply examples/persistence_mongodb.ipynb:338]: all examples that are used in postgres doc have been used now

**Comment by theofficialvedantjoshi** [comment]
when will this be merged?

File: examples/persistence_mongodb.ipynb (hunk 5 above)
Comment (vbarda) [reply examples/persistence_mongodb.ipynb:338]: thanks -- can you show the outputs of `list` (it shows as a generator right now)

File: examples/persistence_mongodb.ipynb (hunk 2 above)
Comment (vbarda) [reply examples/persistence_mongodb.ipynb]: could you also suppress the error outputs from installation in the final notebook? so the errors don't show up in the documentation

**Comment by vbarda** [comment]
> when will this be merged?

added a couple of additional comments, and then should be good to merge

File: examples/persistence_mongodb.ipynb (hunk 5 above)
Comment (theofficialvedantjoshi) [reply examples/persistence_mongodb.ipynb:323]: done

**Comment by vbarda** [comment]
merging -- thanks for adding @theofficialvedantjoshi !

**Comment by vbarda** [comment]
@theofficialvedantjoshi if you could also add async methods to the how-to (`aget_tuple`, `alist`, etc), similar to the postgres one, that would be super helpful!

**Comment by theofficialvedantjoshi** [comment]
> @theofficialvedantjoshi if you could also add async methods to the how-to (`aget_tuple`, `alist`, etc), similar to the postgres one, that would be super helpful!

will create a new pr👍

**Pull request merged**